src/mlpack/core/data/one_hot_encoding_impl.hpp

```diff
@@ -34,36 +34,19 @@ template<typename eT, typename RowType>
 void OneHotEncoding(const RowType& labelsIn,
                     arma::Mat<eT>& output)
 {
-  arma::Row<size_t> labels;
-  labels.set_size(labelsIn.n_elem);
+  RowType labels = arma::unique(labelsIn);
```

Comment:
@jeffin143 I don't think it's reasonable. Suppose you evaluate two times the same formulas. In that case the values might be slightly different due to machine precision issues. Try to calculate the sum `1 / (k * k)` (the series converges) two times:
- from 1 to `N`
- from `N` to 1
and print the result with `std::setprecison(16)`. The last digit may differ (try to choose an appropriate `N`).

How do you think which answer is closer to the value of the infinite series?

However, you expect that the encoded result (with the one hot encoding) should be the same. But, it's not the case since the initial data is different.

Therefore we never use the exact comparison for floating point values. You should introduce a parameter in order to specify the precision of the comparison.